### ISSUE TITLE
Asymmetric matcher utilities broken in v0.28+

### ISSUE THREAD

**Issue by simon-abbott** [vitest-dev/vitest#2744]
### Describe the bug

482b72fc5f02da87498a18e0ed844ecb6bc772ca changed the exported types of [jest-matcher-utils](https://github.com/vitest-dev/vitest/commit/482b72fc5f02da87498a18e0ed844ecb6bc772ca#diff-2d26dc2fba76ea9ac90b604956558afb9a07091abe627b7ac35e8ee6c69f0cb2). The corresponding import in [jest-extend](https://github.com/vitest-dev/vitest/commit/482b72fc5f02da87498a18e0ed844ecb6bc772ca#diff-706e1b952f44f14b9174aef43c867b520c9ab3cc6d91b0268ef6169d87ce9ef6) was changed accordingly, but the one in [jest-asymmetric-matchers](https://github.com/vitest-dev/vitest/blob/8c625d2a10f6a993c7253583bc065aa25c3cdf2a/packages/expect/src/jest-asymmetric-matchers.ts) was not. This causes any asymmetric matchers that use methods like `matcherHint` or `printExpected` to fail.

### Reproduction

Make a custom asymmetric matcher and look at the value of `this.utils`.

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 247.17 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.19 - ~/Programming/Airfordable/ohare/node_modules/.bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
    Watchman: 2023.01.09.00 - /usr/local/bin/watchman
  Browsers:
    Brave Browser: 90.1.24.86
    Chrome: 104.0.5112.101
    Firefox: 109.0
    Firefox Developer Edition: 109.0
    Safari: 16.2
  npmPackages:
    @vitest/ui: 0.28.1 => 0.28.1 
    vitest: 0.28.1 => 0.28.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Vithanco** [comment]
I assume this is the same as my problem: 
```
TypeError: Cannot destructure property 'AsymmetricMatcher' of '__vite_ssr_import_0__.plugins' as it is undefined.
 ❯ node_modules/@vitest/utils/dist/index.js:22:3
     20|   AsymmetricMatcher
     21| ];
     22| function stringify(object, maxDepth = 10, { maxLength, ...options } = {}) {
       |   ^
     23|   const MAX_LENGTH = maxLength ?? 1e4;
     24|   let result;
 ❯ VitestRunner.directRequest node_modules/vite-node/dist/client.mjs:308:5
 ❯ VitestRunner.cachedRequest node_modules/vite-node/dist/client.mjs:156:14
 ❯ VitestRunner.dependencyRequest node_modules/vite-node/dist/client.mjs:206:14
 ❯ async /Users/knownToAuthor/node_modules/@vitest/runner/dist/index.js:3:31
 ❯ VitestRunner.directRequest node_modules/vite-node/dist/client.mjs:308:5
 ❯ VitestRunner.cachedRequest node_modules/vite-node/dist/client.mjs:156:14
 ❯ VitestRunner.dependencyRequest node_modules/vite-node/dist/client.mjs:206:14
 ❯ async /Users/knownToAuthor/node_modules/vitest/dist/entry.js:5:31
 ❯ VitestRunner.directRequest node_modules/vite-node/dist/client.mjs:308:5
```

**Comment by ElkostVN** [comment]
Yes, I faced the same problem
```
Vitest caught 1 unhandled error during the test run.
This might cause false positive tests. Resolve unhandled errors to make sure your tests are not affected.
⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯
TypeError: Cannot destructure property 'AsymmetricMatcher' of '__vite_ssr_import_0__.plugins' as it is undefined.
 ❯ ../../node_modules/@vitest/utils/dist/index.js:22:3
     20|   AsymmetricMatcher
     21| ];
     22| function stringify(object, maxDepth = 10, { maxLength, ...options } = …
       |   ^
     23|   const MAX_LENGTH = maxLength ?? 1e4;
     24|   let result;
 ❯ VitestRunner.directRequest ../../node_modules/vite-node/dist/client.mjs:308:5
 ❯ VitestRunner.cachedRequest ../../node_modules/vite-node/dist/client.mjs:156:14
 ❯ VitestRunner.dependencyRequest ../../node_modules/vite-node/dist/client.mjs:206:14
 ❯ async /home/elkost/pets/server-users-score/node_modules/@vitest/runner/dist/index.js:3:31
 ❯ VitestRunner.directRequest ../../node_modules/vite-node/dist/client.mjs:308:5
 ❯ VitestRunner.cachedRequest ../../node_modules/vite-node/dist/client.mjs:156:14
 ❯ VitestRunner.dependencyRequest ../../node_modules/vite-node/dist/client.mjs:206:14
 ❯ async /home/elkost/pets/server-users-score/node_modules/vitest/dist/entry.js:5:31
 ❯ VitestRunner.directRequest ../../node_modules/vite-node/dist/client.mjs:308:5
```

**Comment by Leksat** [comment]
The [commit](https://github.com/vitest-dev/vitest/commit/415c8a93df3bc688cbc0b40238517ecded7354ed) says the fix was released in v0.28.4, yet I'm still getting the error
```
TypeError: Cannot destructure property 'AsymmetricMatcher' of '__vite_ssr_import_0__.plugins' as it is undefined.
 ❯ ../../../../node_modules/.pnpm/@vitest+utils@0.28.4/node_modules/@vitest/utils/dist/index.js:22:3
     20|   AsymmetricMatcher
     21| ];
     22| function stringify(object, maxDepth = 10, { maxLength, ...options } = {}) {
       |   ^
     23|   const MAX_LENGTH = maxLength ?? 1e4;
     24|   let result;
 ❯ VitestRunner.directRequest ../../../../node_modules/.pnpm/vite-node@0.28.4_@types+node@18.11.18/node_modules/vite-node/dist/client.mjs:312:5
 ❯ VitestRunner.cachedRequest ../../../../node_modules/.pnpm/vite-node@0.28.4_@types+node@18.11.18/node_modules/vite-node/dist/client.mjs:156:14
 ❯ VitestRunner.dependencyRequest ../../../../node_modules/.pnpm/vite-node@0.28.4_@types+node@18.11.18/node_modules/vite-node/dist/client.mjs:204:14
 ❯ async /Users/alex/work/silverback-mono/node_modules/.pnpm/@vitest+runner@0.28.4/node_modules/@vitest/runner/dist/index.js:3:31
 ❯ VitestRunner.directRequest ../../../../node_modules/.pnpm/vite-node@0.28.4_@types+node@18.11.18/node_modules/vite-node/dist/client.mjs:312:5
 ❯ VitestRunner.cachedRequest ../../../../node_modules/.pnpm/vite-node@0.28.4_@types+node@18.11.18/node_modules/vite-node/dist/client.mjs:156:14
 ❯ VitestRunner.dependencyRequest ../../../../node_modules/.pnpm/vite-node@0.28.4_@types+node@18.11.18/node_modules/vite-node/dist/client.mjs:204:14
 ❯ async /Users/alex/work/silverback-mono/node_modules/.pnpm/vitest@0.28.4/node_modules/vitest/dist/entry.js:5:31
 ❯ VitestRunner.directRequest ../../../../node_modules/.pnpm/vite-node@0.28.4_@types+node@18.11.18/node_modules/vite-node/dist/client.mjs:312:5
```

**Comment by sheremet-va** [comment]
> The [commit](https://github.com/vitest-dev/vitest/commit/415c8a93df3bc688cbc0b40238517ecded7354ed) says the fix was released in v0.28.4, yet I'm still getting the error

This is a separate bug not related to the one described in the issue. Please, create a new issue and provide a runnable reproduction.

**Comment by Leksat** [comment]
Thanks! Created https://github.com/vitest-dev/vitest/issues/2806
